File: Retrospective_Core/Models/RetroCard.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Retrospective_Core.Models {
    public class RetroCard {
        
        public int UpVotes { get; set; }

        public int DownVotes { get; set; }

        [ForeignKey("RetroFamilyId")]
        public int? RetroFamilyId { get; set; }

        public virtual RetroFamily RetroFamily { get; set; }

        [Key]
        public int Id { get; set; }

        public string Content { get; set; }

        public int Position { get; set; }

        [ForeignKey("RetroColumnId")]
        public int RetroColumnId { get; set; }
        public virtual RetroColumn RetroColumn { get; set; }
    }
}

File: Retrospective_Front_End/ClientApp/src/models/RetroCard.ts
import { BaseItem } from './BaseItem';

export class RetroCard extends BaseItem {

  constructor(
    public id: number,
    public content: string,
    public position: number,
    public retroColumnId: number,
    public upVotes: number,
    public downVotes: number,
    public familyId: number
  ) {
    super(id, content, position, retroColumnId);
  }
}


